Language/Expressions/Property_Extraction/Instance_Method_Closurization/method_closurization_positional_params_A03_t03.dart

```diff
@@ -37,13 +37,41 @@ class C<X extends num> {
   }
 }
 
+mixin M<X extends num> {
+  X m<Y extends X>(covariant X r1, covariant Y r2, [covariant int? p1]) {
+    return 42 as X;
+  }
+}
+class MO<X extends num> = Object with M<X>;
+
+enum E<X extends num> {
+  e0;
+  X m<Y extends X>(covariant X r1, covariant Y r2, [covariant int? p1]) {
+    return 42 as X;
+  }
+}
+
 main() {
-  var o = C<num>();
-  final f = o.m;
-  f.expectStaticType<
+  var c = C<num>();
+  final fc = c.m;
+  fc.expectStaticType<
         Exactly<num Function<Y extends num>(num r1, Y r2, [int? p1])>
       >();
+  Expect.isTrue(
+    fc is num Function<Y extends num>(Object? r1, Object? r2, [Object? p1]));
```

Comment:
Why wouldn't `r2` have type `Y`? Note that the relaxation of a parameter type to `Object?` which is based on covariance doesn't apply to `Y` because it is a type parameter of the method, not a type parameter of an enclosing class/mixin/etc.

Reply:
Thanks!